src/runtime_src/core/common/api/xrt_xclbin.cpp

```diff
@@ -549,16 +553,26 @@ class xclbin_full : public xclbin_impl
         if (!data.empty()) {
           auto pos = m_axlf_sections.emplace(kind, std::move(data));
           if (kind == IP_LAYOUT)
-            ip_layout = reinterpret_cast<const ::ip_layout*>((pos.first)->second.data());
+            ip_layout = reinterpret_cast<const ::ip_layout*>((pos)->second.data());
         }
       }
 
       if (!hdr)
         continue;
 
-      auto section_data = reinterpret_cast<const char*>(m_top) + hdr->m_sectionOffset;
-      std::vector<char> data{section_data, section_data + hdr->m_sectionSize};
-      m_axlf_sections.emplace(kind , std::move(data));
+      if(kind == SOFT_KERNEL) {
+	while(hdr != nullptr) {
+	  auto section_data = reinterpret_cast<const char*>(m_top) + hdr->m_sectionOffset;
+	  std::vector<char> data{section_data, section_data + hdr->m_sectionSize};
+	  m_axlf_sections.emplace(kind , std::move(data));
+	  hdr = ::xclbin::get_axlf_section_next(m_top, hdr, kind);
+	}
+      }
+      else {
+	auto section_data = reinterpret_cast<const char*>(m_top) + hdr->m_sectionOffset;
+	std::vector<char> data{section_data, section_data + hdr->m_sectionSize};
+	m_axlf_sections.emplace(kind , std::move(data));
+      }
```

Comment:
Do not randomly use tab or spaces.
Configure correctly your text editor or IDE to display them so you can see at least the problem. :-)